### DISCUSSION TITLE
Flux Variablity or Engineering

### DISCUSSION THREAD

**Discussion by zorzansi** [micom-dev/micom#138] [Q&A]
Hi,
we are working on a project involving a microbial community, and we would like to find an optimal flux which remains feasible but maximizes a metabolite or a metabolite(s) class. Do you think that this can be implemented in Micom and how many CPU and RAM do you think such an optimization might require for a community of 6-12 species?

Thank you!

**Comment by cdiener** [comment]
If it's only a handful of metabolites that would be pretty doable. The idea would be to first estimate individual growth rates with cooperative tradeoff for example and then fixing those rates. If it's only for a few metabolites and with a community as small as what you mention this would not take that much CPU and RAM. Like 4GB of RAM and a single core should be fine for that. This would be acceptably fast with free solver and pretty rapid with CPLEX or Gurobi.

As an example, here we maximize the acetate production for the E. coli strain example model:

```python
import micom as mm

com = mm.Community(mm.data.test_taxonomy())
sol = com.cooperative_tradeoff(fraction=1.0)
with com:
    mm.util._apply_min_growth(com, sol.members.growth_rate.drop("medium"))
    com.objective = com.reactions.EX_ac_m
    max_etoh_sol = com.optimize(fluxes=True)
print(max_etoh_sol.fluxes.EX_ac_m)
```
